core/src/main/scala/scalaz/Frap.scala

```diff
@@ -0,0 +1,118 @@
+package scalaz
+
+/**
+ * Free applicative functors. Less expressive than free monads, but more
+ * flexible to inspect and interpret.
+ */
+sealed trait Frap[F[_],A] {
+  import Frap._
+
+  /**
+   * The canonical natural transformation that interprets this free
+   * program by giving it the semantics of the applicative functor `G`.
+   * Not tail-recursive unless `G` is a free monad.
+   */
+  def foldMap[G[_]:Applicative](f: F ~> G): G[A] =
+    this match {
+      case Pure(x) => Applicative[G].pure(x)
+      case x@Ap() => Applicative[G].ap(f(x.v()))(x.k() foldMap f)
+    }
+
+  /** Provides access to the first instruction of this program, if present */
+  def para[B](pure: A => B, ap: ({type λ[α] = (F[α], Frap[F, α => A])})#λ ~>
```

Comment:
That would be awesome. I think you just volunteered to go through the codebase and replace type lambdas with that.